arangod/RocksDBEngine/RocksDBMetadata.cpp

```diff
@@ -470,14 +470,26 @@ Result RocksDBMetadata::deserializeMeta(rocksdb::DB* db, LogicalCollection& coll
     if (!s.ok() && !s.IsNotFound()) {
       return rocksutils::convertStatus(s);
     } else if (s.IsNotFound()) {
-      LOG_TOPIC("ecdbc", WARN, Logger::ENGINES)
-          << "no revision tree found for collection with id '" << coll.id().id()
-          << "', rebuilding";
-      Result res = rcoll->rebuildRevisionTree();
-      if (res.fail()) {
-        LOG_TOPIC("ecdbd", WARN, Logger::ENGINES)
-            << "failed to rebuild revision tree for collection '"
-            << coll.id().id() << "'";
+      // no tree, check if collection is non-empty
+      auto bounds = RocksDBKeyBounds::CollectionDocuments(rcoll->objectId());
+      auto cmp = RocksDBColumnFamily::documents()->GetComparator();
+      std::unique_ptr<rocksdb::Iterator> it{
+          db->NewIterator(ro, RocksDBColumnFamily::documents())};
+      it->Seek(bounds.start());
```

Comment:
This will add an extra `Seek` for each collection/shard at server startup, which could be expensive for very large datasets. Is there a way to avoid this, or at least use the just-created iterator inside `rebuildRevisionTree()`?